### PR TITLE
Fix matching of `@api` class-comments

### PR DESCRIPTION
**Pull request by staabm**
before this PR, any class with just a comment - no matter what content this comment had - was handled like a `@api` commented class

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Comment by staabm** [comment]
@TomasVotruba build should be fixed now :)

**Comment by ruudk** [comment]
Duplicate of https://github.com/TomasVotruba/class-leak/pull/36 ? 😄

File: tests/ClassNameResolver/ClassNameResolverTest.php [hunk 1]
```diff
@@ -25,11 +26,15 @@ protected function setUp(): void
...
     #[DataProvider('provideData')]
-    public function test(string $filePath, ClassNames $expectedClassNames): void
+    public function test(string $filePath, ?ClassNames $expectedClassNames): void
     {
         $resolvedClassNames = $this->classNameResolver->resolveFromFromFilePath($filePath);
-        $this->assertInstanceOf(ClassNames::class, $resolvedClassNames);
+        if (!$expectedClassNames instanceof ClassNames) {
+            $this->assertNull($resolvedClassNames);
+            return;
+        }
```
Comment (TomasVotruba) [inline tests/ClassNameResolver/ClassNameResolverTest.php:35]: I think this should throw an error in this case. We always want a test to provide class.

Comment (staabm) [reply tests/ClassNameResolver/ClassNameResolverTest.php:35]: No. Its a valid case as the file beeing analysed does return null.

Comment (staabm) [reply tests/ClassNameResolver/ClassNameResolverTest.php:35]: @TomasVotruba does this make sense? would be great to get his merged, I have another edge case I want to fix - waiting for this one :)

Comment (staabm) [reply tests/ClassNameResolver/ClassNameResolverTest.php:35]: > I think this should throw an error in this case. We always want a test to provide class.

our test wants to assert that the file beeing analyzed does not return a class (because the only one in the file is annotated `@api`)

Comment (TomasVotruba) [reply tests/ClassNameResolver/ClassNameResolverTest.php:35]: I see. Conditions in test expectations make it hard to read.

Could you split test method for nulls with own data provider?

Comment (staabm) [reply tests/ClassNameResolver/ClassNameResolverTest.php]: good idea. just separated the tests

**Comment by TomasVotruba** [comment]
Awesome :partying_face:  thank you :+1:

**Comment by staabm** [comment]
> I have another edge case I want to fix - waiting for this one :)

ahh its already fixed in https://github.com/TomasVotruba/class-leak/commit/a69ea247434b50de3c685a61d3d797601eb35263

just waiting for a release then :-)

**Comment by TomasVotruba** [comment]
@staabm Better subscribe to releases here, already 2 out :)
